src/librustdoc/core.rs

```diff
@@ -360,9 +376,27 @@ crate fn create_config(
         ..Options::default()
     };
 
+    let crate_cfg = interface::parse_cfgspecs(cfgs);
+
+    let invalid_codeblock_attributes_name = rustc_lint::builtin::INVALID_CODEBLOCK_ATTRIBUTES.name;
+    let broken_intra_doc_links = lint::builtin::BROKEN_INTRA_DOC_LINKS.name;
+
+    let lints_to_show = get_rustdoc_lints(crate_cfg.clone(), sessopts.clone());
+    let (lint_opts, lint_caps) = init_lints(lints_to_show, lint_opts, |lint| {
+        // FIXME: why is this necessary?
+        if lint.name == broken_intra_doc_links || lint.name == invalid_codeblock_attributes_name {
+            None
```

Comment:
If you remove this, does it still warn on those lints?

Reply:
I didn't want to update it in this PR to avoid mixing too many things together. We can take a look when this is merged if you want.